### PR TITLE
Document things

### PR DESCRIPTION
**Pull request by TheBB**
It's difficult to close issues, because so many of them end up in "let's document this". Fortunately, many of them are simple to document.

This PR addresses #1944, #2371, #2476, #2750, #2826, #2851, #3022, #3065, #3181 and #3222, and probably a few things left in #2749 as well (evilification, in particular).

### CODE REVIEW

File: doc/HOWTOs.org [hunk 1]
```diff
@@ -68,3 +69,17 @@ Add the following snippet to your =dostpacemacs/config= function:
     (define-key evil-visual-state-map "j" 'evil-next-visual-line)
     (define-key evil-visual-state-map "k" 'evil-previous-visual-line)
 #+end_src
+
+** Disable evilification of a mode
+You can ensure a mode opens in emacs state by using =evil-set-initial-state=.
+
+#+begin_src emacs-lisp
+  (evil-set-initial-state 'magit-mode 'emacs)
```
Comment (justbur) [inline doc/HOWTOs.org]: @TheBB I don't think this works. Evilify overrides this essentially by setting evilified state as the initial state for more specific magit modes

Comment (TheBB) [inline doc/HOWTOs.org]: Well, it was just an example. I can change it to something that *actually* works.

Comment (TheBB) [inline doc/HOWTOs.org]: Changed to `magit-status-mode`.

Comment (justbur) [inline doc/HOWTOs.org]: how about this?
```elisp
(add-to-list 'evil-buffer-regexps '("*magit" . emacs))
```

**Comment by justbur** [comment]
Added this to an old diff, so you might have missed it. This is a blunt way to prevent evilified state in all magit buffers

```elisp
(add-to-list 'evil-buffer-regexps '("*magit" . emacs))
```

**Comment by TheBB** [comment]
I saw it, but the tip wasn't about magit specifically, I just used that as an example of an evilified major mode.

**Comment by TheBB** [comment]
Does that method take precedence over the mode lists?

**Comment by justbur** [comment]
Yes it does. I think magit is the leading case, and with magit there are many modes that you have to override.

**Comment by TheBB** [comment]
Okay, updated.

**Comment by justbur** [comment]
Excellent :+1:

**Comment by TheBB** [comment]
Added a few more issues.

**Comment by TheBB** [comment]
I trust @justbur's review, so I just merged this.

File: doc/DOCUMENTATION.org [hunk 2]
```diff
@@ -642,6 +651,24 @@ in =evil-hybrid-state-map= keymap instead of =evil-insert-state-map=.
...
+editing text, and provide their own keybindings for certain operations. These
+often conflict with Vim bindings. To make such buffers behave Vim-like in a
+consistent manner, they use a special state called /evilified/ state. In
+evilified state, a handful of keys work as in Evil, namely =/=, =:=, =h=, =j=,
+=k=, =l=, =n=, =N=, =v=, =V=, =gg=, =G=, =C-f=, =C-b=, =C-d=, =C-u= and =C-z=.
+All other keys work as intended by the underlying mode.
+
+Shadowed keys are moved according to the pattern: =a= → =A= → =C-a= → =C-A=
+
+For example, if the mode binds a function to =n=, that is found under =C-n= in
```
Comment (syl20bnr) [inline doc/DOCUMENTATION.org]: @TheBB It will be bound to `N` instead of `C-n`

Comment (syl20bnr) [inline doc/DOCUMENTATION.org]: Oh actually it is `C-n` but you should explain again why just to be explicit and make sure that the reader understood.

**Comment by syl20bnr** [comment]
Very nice PR :+1:

File: doc/DOCUMENTATION.org (hunk 2 above)
Comment (TheBB) [inline doc/DOCUMENTATION.org]: Sure thing!

Comment (TheBB) [inline doc/DOCUMENTATION.org]: https://github.com/syl20bnr/spacemacs/commit/91947406f8901c19a38db174f3704c5d66bf6de1

**Pull request closed without merging**